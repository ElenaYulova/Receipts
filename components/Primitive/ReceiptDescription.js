import React from 'react';
import PropTypes from 'prop-types';

import './ReceiptDescription.css';

// Компонент Описание рецепта кулинарной книги. Переиспользуемый
export default class ReceiptDescription extends React.PureComponent {
    static propTypes = {
        
        description: PropTypes.string.isRequired,
        
    };

    
    render() {
        
        const text = this.props.description;
    return <div  className="receipt-description">
        {text}        
    </div>
 }   
}