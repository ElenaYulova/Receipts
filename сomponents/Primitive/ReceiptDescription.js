import React from 'react';
import PropTypes from 'prop-types';

import './ReceiptDescription.css';

// Компонент Описание рецепта кулинарной книги. Переиспользуемый
export default class ReceiptDescription extends React.PureComponent {
    static propTypes = {
        name: PropTypes.string.isRequired,
        description: PropTypes.string.isRequired,
        
    };

    
    render() {
        const name = this.props.name;
        const text = this.props.description;
    return <div id = {name} className="receipt-description">
        {text}        
    </div>
 }   
}