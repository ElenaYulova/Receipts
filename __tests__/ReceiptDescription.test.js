"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import ReceiptDescription from '../components/Primitive/ReceiptDescription';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });

describe('>>>ReceiptDescription --- Shallow Render REACT COMPONENTS',()=>{
    //shapshots
     
     
      const receiptComponent = <ReceiptDescription 
      name = "name1"
      description = "ADSDfvSDvsD"
       />;
    
         
         const component = renderer.create(receiptComponent);
    
         it('renders receipt', () => {
           let componentTree = component.toJSON();
           expect(componentTree).toMatchSnapshot();
         });
    
         
    });