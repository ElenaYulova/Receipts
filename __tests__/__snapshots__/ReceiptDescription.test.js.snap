// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>ReceiptDescription --- Shallow Render REACT COMPONENTS renders receipt 1`] = `
<div
  className="receipt-description"
>
  ADSDfvSDvsD
</div>
`;
